src/Service/GameService.php

```diff
@@ -151,6 +154,25 @@ public function rotateQuarterBy90Degrees(Game $game, int $rotationKey): Game
         // Player has finished his turn
         $game->setCurrentPlayerHash($this->changeCurrentPlayerHash($game));
 
+        // At this step we have also to check if game is finished.
+        $game = $this->checkIfGameIsFinished($game);
+
+        return $game;
+    }
+
+    public function checkIfGameIsFinished(Game $game): Game
+    {
+        $results = $this->winDetectionService->getAllMarblesCombinationsCorrectlyAligned($game->getBoard());
+
+        $numberOfWinners = count($results["winners"]);
+        if ($numberOfWinners > 0) {
+            $game->setStatus(self::GAME_FINISHED);
+            $game->setAllAlignedPositions($results["allAlignedPositions"]);
+
+            $winner = $numberOfWinners === 1 ? array_key_first($results["winners"]) : 0;
```

Comment:
`array_key_first` didn't know about this one

Reply:
$results["winners"] is something like :
```
[
  "1" => true,
  "2" => true
]
```
I actually don't like this field and it will be changed in the next PR but array_key_first allow me to get the first key in this associative array.
Because in this case $results["winners"][0] doesn't exist.